src/Graph.ts

```diff
@@ -1,67 +1,24 @@
 import { Entity, Value } from '@alephdata/followthemoney'
 import { Vertex } from './Vertex'
 import { Edge } from './Edge'
+import { Viewport } from './Viewport'
 import { EntityVertex } from './EntityVertex'
 import { ValueVertex } from './ValueVertex'
 import { EntityEdge } from './EntityEdge'
 import { PropertyEdge } from './PropertyEdge'
-import { Point } from './Point'
 
-
-export interface IGraphEvent {
-  vertices: Array<Vertex>,
-  edges: Array<Edge>,
-  zoomFactor: number,
-  panCenter: Point
-}
-
-export type GraphEventListener = (event: IGraphEvent) => void
+export type GraphUpdateHandler = (graph: Graph) => void
 
 export class Graph {
-  panCenter = new Point({x:0, y:0})
-  zoomFactor = 4;
+  viewport: Viewport;
   vertices: Map<string, Vertex> = new Map()
   edges: Map<string, Edge> = new Map()
   entities: Map<string, Entity> = new Map()
-  listeners : Set<GraphEventListener> = new Set();
 
   constructor() {
+    this.viewport = new Viewport(1)
```

Comment:
I would suggest initializing `viewport` in class-properties if they don't depend from `arguments` of constructor

Reply:
+1